src/main/java/net/masterthought/cucumber/ReportBuilder.java

```diff
@@ -98,6 +110,30 @@ private void generateAllPages() {
         }
 
         new StepsOverviewPage(reportResult, configuration).generatePage();
+
+        try {
+            saveAttachments();
+        } catch (IOException e) {
+            new ValidationException(e);
+        }
+    }
+
+    private void saveAttachments() throws IOException {
+        File path = new File(
+                configuration.getReportDirectory().getAbsolutePath(), EMBEDDINGS_DIRECTORY);
+        path.mkdirs();
+        for (Feature feature : reportResult.getAllFeatures()) {
+            for (Element element : feature.getElements()) {
+                for (Step step : element.getSteps()) {
+                    for (Embedding embedding : step.getEmbeddings()) {
+                        Path file = FileSystems.getDefault().getPath(
+                                path.getAbsolutePath(), embedding.hashCode() + "." + embedding.getExtension());
+                        Files.write(file, Base64.decodeBase64(embedding.getData().getBytes()));
```

Comment:
In your tests you use `getDecodedData` which properly provides `Charsets.UTF_8`; however, here you're not passing `Charsets.UTF_8` to the `getBytes()` call.  I think that may explain why the embeddings are being saved in a format my browser can't understand.